compiler/rustc_resolve/src/build_reduced_graph.rs

```diff
@@ -210,6 +210,17 @@ impl<'ra, 'tcx> Resolver<'ra, 'tcx> {
         }
     }
 
+    /// Add every proc macro accessible from the current crate to the `macro_map` so diagnostics can
+    /// find them for suggestions.
+    pub(crate) fn register_macros_for_all_crates(&mut self) {
```

Comment:
The code this is replacing would register any non-proc-macro derives or attributes, if any; this seems to only register proc macros. With the addition of `attr` and `derive` rules for `macro_rules!` macros, this should be registering non-proc-macros for possible suggestions too.